src/main/target/HELIOSPRING/target.h

```diff
@@ -0,0 +1,205 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define TARGET_BOARD_IDENTIFIER "HESP"
+#define USBD_PRODUCT_STRING     "HELIOSPRING"
+#define UPT_ADDRESS  0x080F0000
+#define MSD_ADDRESS  0x080E0000
+
+#define LED0_PIN                PB7
+
+#define USE_BEEPER
+#define BEEPER_PIN                  PC15
+#define BEEPER_INVERTED
+
+#define USE_GYRO
+#define USE_ACC
+
+#define DEFAULT_ATTITUDE_UPDATE_INTERVAL 1000 
+#define DEFAULT_ACC_SAMPLE_INTERVAL      1000
+
+#define USE_FAST_SPI_DRIVER
+#define USE_GYRO_IMUF9001
+#define USE_QUAT_IMUF9001
+#define USE_ACC_IMUF9001
+#define IMUF9001_CS_PIN         PB1
+#define IMUF9001_RST_PIN        PA4 
+#define IMUF9001_SPI_INSTANCE   SPI1
+#define USE_EXTI
+#define MPU_INT_EXTI            PB0
+#define USE_MPU_DATA_READY_SIGNAL
+
+
+#define USE_DSHOT_DMAR
+#define ENABLE_DSHOT_DMAR       true
+
+
+#define USE_FLASH
+#define USE_FLASHFS
+#define USE_FLASH_M25P16
+#define FLASH_SPI_INSTANCE      SPI3
+#define FLASH_CS_PIN            SPI3_NSS_PIN
+#define ENABLE_BLACKBOX_LOGGING_ON_SPIFLASH_BY_DEFAULT
+
+#define USE_VCP
+#define USE_USB_DETECT
+#define USB_DETECT_PIN          PC5
+
+#define USE_OSD
+#define USE_MAX7456
+#define MAX7456_SPI_INSTANCE    SPI3
+#define MAX7456_SPI_CS_PIN      PA15
+#define MAX7456_SPI_CLK         (SPI_CLOCK_STANDARD) // 10MHz
+#define MAX7456_RESTORE_CLK     (SPI_CLOCK_FAST)
+
+#define USE_UART1
+#define UART1_RX_PIN            PA10
+#define UART1_TX_PIN            PA9
+
+#define USE_UART2
+#define UART2_RX_PIN            PA3
+#define UART2_TX_PIN            PA2
+
+#define USE_UART3
+#define UART3_RX_PIN            PB11
+#define UART3_TX_PIN            PB10
+
+#define USE_UART4
+#define UART4_RX_PIN            PC11
+
+#define USE_UART5
+#define UART5_RX_PIN            PD2
+#define UART5_TX_PIN            PC12
+
+#define SERIAL_PORT_COUNT       6 //VCP, USART1, USART2, USART3, UART4, USART5
+
+#define USE_SPI
+
+#define USE_SPI_DEVICE_1
+#define SPI1_NSS_PIN            PB1
+#define SPI1_SCK_PIN            PB3
+#define SPI1_MISO_PIN           PA6
+#define SPI1_MOSI_PIN           PA7
+
+#define USE_DMA_SPI_DEVICE
```

Comment:
`DMA_SPI` needs to be abstracted out and made configurable just like the entire IO facility.